### ISSUE TITLE
make new features/changelog more visible to users

### ISSUE THREAD

**Issue by dixler** [pulumi/pulumi#10090]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

We write to `CHANGELOG.md`, but if users are upgrading via our recommendation in the CLI they don't see it.

---
```
$ pulumi version
```
output:
```
v3.30.0
warning: A new version of Pulumi is available. To upgrade from version '3.30.0' to '3.35.3', run 
   $ curl -sSL https://get.pulumi.com | sh
or visit https://pulumi.com/docs/reference/install/ for manual instructions and release notes.
```

If users aren't made aware that previous limitations/restrictions are now lifted, they will continue to assume those restrictions are still there.

#### Suggestions

Modify the upgrade message to include a link to the CHANGELOG.md
```
v3.30.0
warning: A new version of Pulumi is available. To upgrade from version '3.30.0' to '3.35.3', run 
   $ curl -sSL https://get.pulumi.com | sh
or visit https://pulumi.com/docs/reference/install/ for manual instructions.

Release notes:
https://github.com/pulumi/pulumi/blob/master/CHANGELOG.md
```

### Affected area/feature

Ideas

- display changelog on upgrade

**Comment by dixler** [comment]
Note: converted this to an issue from an ideation board so it looks unconventional.

**Comment by dixler** [comment]
duplicate of #8764

**Comment by dixler** [comment]
reopening since #8764 has 2 feature requests and this is one of them so I will edit the issue to link back to this issue as a work item.

**Comment by dixler** [comment]
Ideas:
- [ ] with the upgrade script we could add a permalink to the changelog with new lines highlighted

**Comment by dixler** [comment]
notes: @meagancojocar has a format to disseminate new features to users.

**Comment by Frassle** [comment]
I love the idea of making the changelog more discoverable and easier for users to notice. I wonder if we can tie these thought in while looking at https://github.com/pulumi/pulumi/issues/9945?

**Comment by mikhailshilkov** [comment]
To make this great, we roughly need to do the following:

- Track when the CLI got updated and from which version. Since we use multiple distribution channels, it's easier to make this a core feature of the CLI itself. For example, save the last used version in `~/.pulumi/version` or similar, then compare at startup
- Only show the hint once, then update `~/.pulumi/version` again, so that the check doesn't yield another message next time
- Have some way to give more weight to important changes. Have some star-rating in the changelog itself? Anything we blogged about - add links to the blog retrospectively?
- Have a way to retrieve and condense the changelog between version A and B. We likely don't want to add the entire changelog printout if a user updates multiple versions, but give a summary of most important features. Filter by A < x < B, sort by importance, and return top N, or call an LLM to summarize the result.

**Comment by tgummerer** [comment]
> call an LLM to summarize the result.

Would we trust it enough to not hallucinate non-existant features?  I'd worry that even one wrong changelog entry causing confusion would be much worse than not showing the changelog at all.

> Have some way to give more weight to important changes. Have some star-rating in the changelog itself? Anything we blogged about - add links to the blog retrospectively?

Imo if we want to show this on the CLI rather than just adding a link to the changelog, we should probably have a summary that's manually generated with each release, potentially linking to the `CHANGELOG.md` for more information.  Alternatively we can also introduce something like `pulumi news` that shows all the latest changes, and instead of linking just advertise that command (especially if there are more new things than we consider reasonable to show to a user)

**Comment by mikhailshilkov** [comment]
> Would we trust it enough to not hallucinate non-existant features? I'd worry that even one wrong changelog entry causing confusion would be much worse than not showing the changelog at all.

Yes, I think we can trust it here. They are pretty good at text summarization at this point, this is a rather trivial task.

> we should probably have a summary that's manually generated with each release

What do you mean by "manually generated" here? We already write manual entries - do you mean highlighting the most important points?

How do you produce a summary across multiple releases? (somebody upgrading 160 to 165 or 140 to 165)

**Comment by tgummerer** [comment]
> What do you mean by "manually generated" here? We already write manual entries - do you mean highlighting the most important points?

We do write manual entries, but it's hard to rate the importance of them consistently before a release imo.  The changelog entries that we display like this probably deserve some extra care since they are shown much more often, so I think it's worth manually summarizing things at release time yes. 

> How do you produce a summary across multiple releases? (somebody upgrading 160 to 165 or 140 to 165)

Just show the notes for the latest, and point users to either the changelog or `pulumi latest-changes` or something similar.  Maybe an LLM can summarize things for 25 releases, but it's probably not going to know which ones we are prioritizing.  And having priority across 25 releases and only picking the top say 5 entries is also going to be very hard to do consistently.

**Comment by mikhailshilkov** [comment]
> Just show the notes for the latest

This leaves out too much value: a random release will have 0 to 1 updates worth highlighting. I would say the feature is not worth the effort in this case. Either we make it great and highlight exactly what we want to highlight or we don’t do it at all.

Another aspect that just came to my mind. We used to do “release notes” blog posts that were manually curated. There is probably lots of overlap between those and the CLI feature.

**Comment by tgummerer** [comment]
>  random release will have 0 to 1 updates worth highlighting. I would say the feature is not worth the effort in this case. Either we make it great and highlight exactly what we want to highlight or we don’t do it at all.

I would have expected more than that, but I guess it depends what we consider worth highlighting.  If we only have that few we can probably show all of them then without it getting too long even for upgrades?
